src/actions/analytics.ts

```diff
@@ -171,7 +174,18 @@ export function trackAction(action: string, data: Record<string, unknown>) {
     (acc, key) => ({ ...acc, [key]: data[key] ? data[key] : '1' }),
     {},
   );
-
+  // Format event data to correct firebase format
+  const firebaseContextData = Object.keys(data).reduce(
+    (acc, key) => ({
+      ...acc,
+      [key.replace(/[. ]/g, '_').toLowerCase()]: data[key] ? data[key] : '1',
```

Comment:
```suggestion
      [key.replace(/[. ]/g, '_').toLowerCase()]: data[key] || '1',
```
or
```suggestion
      [key.replace(/[. ]/g, '_').toLowerCase()]: data[key] ?? '1',
```
depending on if you want falsy things other than `undefined` and `null` passed through.